klippy/extras/x_twist_compensation.py

```diff
@@ -0,0 +1,633 @@
+        # remove the profile from self.profiles
+        profiles = dict(self.profiles)
+        del profiles[profile_name]
+        self.profiles = profiles
+        # inform user to save deletion
+        self.gcode.respond_info(
+            "Profile [%s] removed from storage for this session.\n"
+            "The SAVE_CONFIG command will update the printer\n"
+            "configuration and restart the printer" % (profile_name))
+
+    def _save_profile(self, profile):
+        profile_name = profile.name
+        config_name = '%s %s' % (self.name, profile_name)
+        configfile = self.printer.lookup_object('configfile')
+        # save the profile to config file
+        # also save to self.profiles, make sure immutable by making a copy
+        profiles = dict(self.profiles)
+        profiles[profile_name] = new_profile = {}
+        for option, option_type in Profile.PROFILE_OPTIONS.items():
+            value = getattr(profile, option)
+            if option_type == float:
+                value = float(value)
+                new_profile[option] = value  # save to self.profiles
+                configfile.set(config_name, option,
+                               Helpers.format_float_to_n_decimals(value))
+            elif option_type == str:
+                if option == 'z_compensations':
+                    # convert to list of floats
+                    value = [float(x) for x in value]
+                    new_profile[option] = value  # save to self.profiles
+                    value_as_str = [Helpers.format_float_to_n_decimals(
+                        x) for x in value]  # convert to list of strs
+                    configfile.set(config_name, option, ', '.join(
+                        value_as_str))  # store as comma separated
+                else:
+                    new_profile[option] = value  # save to self.profiles
+                    configfile.set(config_name, option, value)
+        # inform user to save changes
+        self.gcode.respond_info(
+            "X_TWIST_COMPENSATION state has been saved to profile [%s]\n"
+            "for the current session.  The SAVE_CONFIG command will\n"
+            "update the printer config file and restart the printer."
+            % (profile_name))
+
+    def save_current_profile(self, profile_name):
+        # get the current profile
+        profile = self.get_current_profile()
+        # set the name
+        profile.name = profile_name
+        # save the profile
+        self._save_profile(profile)
+
+    cmd_X_TWIST_PROFILE_LOAD_help = \
+        "Loads a saved mesh as the active mesh"
+
+    def cmd_X_TWIST_PROFILE_LOAD(self, gcmd):
+        # loads a saved mesh as the active mesh
+        profile_name = gcmd.get('NAME', None)
+        Helpers.check_non_empty_param(profile_name, gcmd, 'NAME')
+        self.load_profile(profile_name)
+
+    cmd_X_TWIST_PROFILE_CLEAR_help = \
+        "Clears the active mesh"
```

Comment:
Is mesh the correct wording here?
```suggestion
        "Clears the active X twist compensation"
```